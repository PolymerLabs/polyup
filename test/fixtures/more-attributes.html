<link rel="import" href="../polymer/polymer.html">

<polymer-element name="attributes-element" attributes="a b c"
                 x="1" y="2" on-scroll='{{scrollFunc}}'>
  <script>
    Polymer({
      a: 10,
      d: true,
      e: [],
      observe: {
        'f g': 'fAndGChanged'
      },
      computed: {
        h: '1 + 1',
        i: 'a + b',
        j: 'callOk(b, c + h)'
      }
    })
  </script>
</polymer-element>
