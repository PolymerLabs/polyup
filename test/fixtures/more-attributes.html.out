<link rel="import" href="../polymer/polymer.html">

<dom-module id="attributes-element">
  <script>
    Polymer({
      is: 'attributes-element',
      properties: {
        a: {
          type: Number,
          value: 10,
          notify: true
        },
        b: { notify: true },
        c: { notify: true },
        d: {
          type: Boolean,
          value: true
        },
        e: {
          type: Array,
          value: function () {
            return [];
          }
        },
        f: { observer: 'fAndGChanged' },
        g: { observer: 'fAndGChanged' },
        h: { computed: 'computeH()' },
        i: { computed: 'computeI(a, b)' },
        j: { computed: 'computeJ(b, c, h)' }
      },
      hostAttributes: {
        'x': '1',
        'y': '2'
      },
      listeners: { scroll: 'scrollFunc' },
      computeH: function () {
        return 1 + 1;
      },
      computeI: function (a, b) {
        return a + b;
      },
      computeJ: function (b, c, h) {
        return this.callOk(b, c + h);
      }
    });
  </script>
</dom-module>
